pkg/core/mempool/hashmap.go

```diff
@@ -110,6 +110,13 @@ func (m *HashMap) Contains(txID []byte) bool {
 	return ok
 }
 
+// Get returns a tx for a given txID if it exists.
+func (m *HashMap) Get(txID []byte) transactions.Transaction {
+	var k txHash
+	copy(k[:], txID)
+	return m.data[k].tx
```

Comment:
Could we check here if key exists with ` txdesc, ok := m.data[k]`. Currently, it does not panic as map contains values but if someone changes it to map of references (like `data map[txHash]*TxDesc`) , this would be a hidden vulnerability making mempool panic.

Reply:
Good point